### PR TITLE
Fire TSP client requests in parallel in getData function

### PR DESCRIPTION
**Pull request by williamsyang-work**
Requests for States, Annotations, and Arrows are now fired simultaneously.

Annotation response data processing moved in front of states processing, since annotation response time is usually shorter.

Signed-off-by: Will Yang <william.yang@ericsson.com>

### CODE REVIEW

**Review by MatthewKhouzam — DISMISSED** [review]
LGTM, but I asked for it.

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts [hunk 1]
```diff
@@ -48,106 +49,113 @@ export class TspDataProvider {
...
-                categoryArray.forEach(annotation => {
-                    if (annotation.type === Type.CHART) {
-                        if (annotation.entryId === -1) {
-                            rangeEvents.push(this.getAnnotation(category, annotation, rangeEvents.length, chartStart));
-                        } else {
-                            let entryArray = annotations.get(annotation.entryId);
-                            if (entryArray === undefined) {
-                                entryArray = [];
-                                annotations.set(annotation.entryId, entryArray);
+            const tspClientAnnotationsResponse = await annotationsPromise;
```
Comment (bhufmann) [inline packages/react-components/src/components/data-providers/tsp-data-provider.ts:86]: when executing this await, all 3 created promises (for annotations, states and arrows) are triggered and the back-end requests are done. With this all 3 are sent in parallel. However, we should make it more explicit and schedule all 3 executions using Promise.all() or Promise.allSettled().

**Review by bhufmann — CHANGES_REQUESTED** [review]

**Comment by bhufmann** [comment]
@williamsyang-work any news on this one?

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts [hunk 2]
```diff
@@ -60,48 +56,38 @@ export class TspDataProvider {
...
-        const rangeEvents: TimelineChart.TimeGraphAnnotation[] = [];
+        const annotationsPromise = this.client.fetchAnnotations(this.traceUUID, this.outputId, annotationParams);
+
+        const arrowStart = viewRange.start + this.timeGraphEntries[0].start;
+        const arrowEnd = viewRange.end + this.timeGraphEntries[0].start;
+        const fetchParameters = QueryHelper.timeRangeQuery(arrowStart, arrowEnd, nbTimes);
+        const arrowsPromise = this.client.fetchTimeGraphArrows(this.traceUUID, this.outputId, fetchParameters);
+
+        // Wait for responses
+        const [tspClientAnnotationsResponse, tspClientStatesResponse, tspClientArrowsResponse] = [await annotationsPromise, await statesPromise, await arrowsPromise];
```
Comment (bhufmann) [inline packages/react-components/src/components/data-providers/tsp-data-provider.ts:83]: Use the following Promise.all() to make sure that the 3 promises are triggered at the same time:
`const [tspClientAnnotationsResponse, tspClientStatesResponse, tspClientArrowsResponse] = await Promise.all([annotationsPromise, statesPromise, arrowsPromise]);`

**Review by bhufmann — CHANGES_REQUESTED** [review]

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks for this improvement.

**Review by MatthewKhouzam — APPROVED** [review]
Looks good to me!

**Pull request merged**